web/pdf_page_view.js

```diff
@@ -653,35 +664,33 @@ class PDFPageView {
             textLayer.setTextContentStream(readableStream);
             textLayer.render();
           }
+
+          if (this._annotationCanvasMap) {
+            if (!this.annotationLayer) {
+              this.annotationLayer =
+                this.annotationLayerFactory.createAnnotationLayerBuilder(
+                  div,
+                  pdfPage,
+                  /* annotationStorage = */ null,
+                  this.imageResourcesPath,
+                  this._annotationMode === AnnotationMode.ENABLE_FORMS,
+                  this.l10n,
+                  /* enableScripting = */ null,
+                  /* hasJSActionsPromise = */ null,
+                  /* mouseState = */ null,
+                  /* fieldObjectsPromise = */ null,
+                  /* annotationCanvasMap */ this._annotationCanvasMap
+                );
+            }
+            this._renderAnnotationLayer();
+          }
         });
       },
       function (reason) {
         return finishPaintTask(reason);
       }
     );
 
-    if (
-      this._annotationMode !== AnnotationMode.DISABLE &&
-      this.annotationLayerFactory
-    ) {
-      if (!this.annotationLayer) {
-        this.annotationLayer =
-          this.annotationLayerFactory.createAnnotationLayerBuilder(
-            div,
-            pdfPage,
-            /* annotationStorage = */ null,
-            this.imageResourcesPath,
-            this._annotationMode === AnnotationMode.ENABLE_FORMS,
-            this.l10n,
-            /* enableScripting = */ null,
-            /* hasJSActionsPromise = */ null,
-            /* mouseState = */ null,
-            /* fieldObjectsPromise = */ null
-          );
-      }
-      this._renderAnnotationLayer();
-    }
-
```

Comment:
You shouldn't need to move all of this code, *only* the `this._renderAnnotationLayer();` line as far as I can tell!?